Makefile

```diff
@@ -148,6 +149,7 @@ E2E_TEMPLATES_DIR ?= $(ROOT_DIR)/test/e2e/data/infrastructure-metal3
 cluster-templates: $(KUSTOMIZE) ## Generate cluster templates
 	$(KUSTOMIZE) build $(E2E_TEMPLATES_DIR)/cluster-template-ubuntu > $(E2E_OUT_DIR)/cluster-template-ubuntu.yaml
 	$(KUSTOMIZE) build $(E2E_TEMPLATES_DIR)/cluster-template-centos > $(E2E_OUT_DIR)/cluster-template-centos.yaml
+	$(KUSTOMIZE) build $(E2E_TEMPLATES_DIR)/cluster-template-upgrade-workload --load-restrictor LoadRestrictionsNone > $(E2E_OUT_DIR)/cluster-template-upgrade-workload.yaml
```

Comment:
I really don't like that we need `--load-restrictor` and I have a suggestion for how to get rid of it. It is needed only when trying to reference files outside of the working directory for patches, so we can move things around a bit to fix it.

```
.
└── infrastructure-metal3
    ├── bases
    │   ├── centos-kubeadm-config    # Put the kubeadm patches here but do NOT include the IP pool.
    │   ├── cluster
    │   ├── ippool
    │   └── ubuntu-kubeadm-config   # Put the kubeadm patches here but do NOT include the IP pool.
    ├── cluster-template-centos         # Here we put a kustomization that uses ippool and kubeadm-config from bases
    ├── cluster-template-ubuntu         # Here we put a kustomization that uses ippool and kubeadm-config from bases
    └── cluster-template-ubuntu-workload  # Here kustomization that uses only kubeadm-config
```